internal/codespaces/portforwarder/port_forwarder.go

```diff
@@ -128,8 +164,57 @@ func (fwd *PortForwarder) ForwardPort(ctx context.Context, remotePort uint16, li
 	}
 }
 
+// connectToForwardedPort connects to the forwarded port via a local TCP port or a given ReadWriteCloser.
+// Optionally, it detects traffic over the connection and sends activity signals to the server to keep the codespace from shutting down.
+func (fwd *CodespacesPortForwarder) connectToForwardedPort(ctx context.Context, portNumber uint16, keepAlive bool, listener *net.TCPListener, conn io.ReadWriteCloser) (err error) {
```

Comment:
Passing around both the TCPListener and the ReadWriteCloser is a little awkward given that only one (should?) ever be set. I wonder if we should just make a separate method for using the ReadWriteCloser to short circuit it straight into the `ConnectToForwardedPort` below, rather than keep them on a shared path - WDYT?

Reply:
I agree that it's a bit awkward, updated things to avoid that 🙂